Marksman/Server.fs

```diff
@@ -715,4 +722,53 @@ type MarksmanServer(client: MarksmanClient) =
 
         AsyncLspResult.success tokens
 
+    override this.TextDocumentCodeAction(opts: CodeActionParams) =
+        let state = requireState ()
+        let docPath = opts.TextDocument.Uri |> PathUri.fromString
+        let range = opts.Range
+
+        let p = State.tryFindDocument docPath state
```

Comment:
Perhaps match on the option?
```
let action = 
  match State.tryFindDocument docPath state with
  | None  ...
  | Some ...
```

Or you could use a `monad { }` [computation expression](https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/computation-expressions) which is like for comprehension or do notation:
```
let action =
  monad {
    let! doc = State.tryFindDoc docPath state
    let toc = Toc.mk doc.index
  }
```